backend/src/error.rs

```diff
@@ -0,0 +1,62 @@
+use actix_web::{error::ResponseError, http::StatusCode, HttpResponse};
+use serde::Serialize;
+use thiserror::Error;
+
+#[derive(Serialize)]
+pub struct ErrorResponse {
+    code: u16,
+    error: String,
+    message: String,
+}
+
+#[derive(Debug, Error, Serialize)]
+pub enum HiveError {
```

Comment:
nit: maybe rename to `ServerError`, since these errors might have nothing to do with a Hive game